src/Raven.Client/Json/Converters/StreamConverter.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Raven.Client.Extensions.Streams;
+using Sparrow.Json;
+
+namespace Raven.Client.Json.Converters
+{
+    internal sealed class StreamConverter : RavenJsonConverter
+    {
+        public static readonly StreamConverter Instance = new StreamConverter();
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (!(writer is BlittableJsonWriter blittableJsonWriter))
+            {
+                throw new SerializationException($"{nameof(BlittableJsonWriter)} must to be used for serialize {nameof(Stream)}");
+            }
+
+            if (!(value is Stream stream))
+            {
+                throw new SerializationException($"Try to write non {nameof(Stream)} value with {nameof(StreamConverter)}");
+            }
+
+            //Todo To consider if Seek need to be here or should it come ready
```

Comment:
You do need it here, and you need to check that you `CanSeek` and error appropriately.
Because you consume the full data, the serialization mutates the state of the object. We want to return it to `0` after the call.

Reply:
done